DQM code for path HLT_TripleMu_5_3_3_Mass3p8to60_DZ

gkaratha opened the pull request:
Dear DQM experts,

THis is the DQM code for 3ple muon extension of our analysis. It includes 3 lines more in the code. It belongs to the SUSY PAG. The code is an addition of topMonitoring.

gkaratha committed:
SOS 3ple muon dqm

cmsbuild commented:
A new Pull Request was created by  @gkaratha  for master.

It involves the following packages:

DQMOffline/Trigger


@kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20212/206

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20212/206/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20212/206/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

mtosi commented on DQMOffline/Trigger/python/SusyMonitoring_Client_cff.py:
```diff
@@ -98,6 +98,38 @@
+    verbose        = cms.untracked.uint32(0), # Set to 2 for all messages 
+    resolution     = cms.vstring(),
+    efficiency     = cms.vstring(
+      "effic_Mll       'efficiency vs inv.mass; Mll [GeV]; efficiency' invMass_numerator       invMass_denominator",
```
do you need it ?

mtosi commented on DQMOffline/Trigger/python/SusyMonitoring_cff.py:
```diff
@@ -157,6 +157,32 @@
+
+# Triggers
+triple_muon_mupt.numGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_TripleMu_5_3_3_Mass3p8to60_DZ_v*')
+triple_muon_mupt.denGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_Trimuon2_Upsilon5_Muon_v*')
```
in which PD are you going to have this monitoring ?

rcastello replied:
@mtosi  I think the DoubleMuon PD seems the most appropriate for this path. Thanks.

mtosi replied:
can you please check the name of this trigger path ?
HLT_Trimuon2_Upsilon5_Muon_v

in addition, it would be useful to see some (filled) plots produced by your code ...

mtosi commented on DQMOffline/Trigger/python/SusyMonitoring_cff.py:
```diff
@@ -157,6 +157,32 @@
+triple_muon_mll.FolderName   = cms.string('HLT/SUSY/SOS/TripleMu/Mll/')
+# Selections
+triple_muon_mll.nmuons           = cms.uint32(3)
+triple_muon_mll.muoSelection     =cms.string('pt>5 & abs(eta)<2.4')
```
are you not applying any ID selection ?

mtosi commented on DQMOffline/Trigger/plugins/TopMonitor.cc:
```diff
@@ -590,6 +600,22 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
+    std::vector<TLorentzVector> mu;
+    for (unsigned int idx=0; idx<muons.size(); idx++){
+      TLorentzVector temp_mu;
+      temp_mu.SetPtEtaPhiM(muons[idx].pt(),muons[idx].eta(),muons[idx].phi(),0.1);
```
can you please either use 0 or the m_mu ?

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

mtosi commented:
tracked by https://github.com/cms-sw/cmssw/issues/19142

cmsbuild commented:
+1
Tested at: a4c96ff5124891f12f395b71f689325cf42ab663
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22385/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/40a53de7f35e63c79a2f076845700fc5182bddd6
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22385/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22385/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22385/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 211
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653534
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

mtosi commented:
is

HLT_Trimuon2_Upsilon5_Muon_v

available in that PD ?


On Mon, Aug 21, 2017 at 12:18 PM, Roberto Castello <notifications@github.com
> wrote:

> *@rcastello* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/python/SusyMonitoring_cff.py
> <https://github.com/cms-sw/cmssw/pull/20212#discussion_r134195380>:
>
> > @@ -157,6 +157,32 @@
>  double_soft_muon_backup_90_mhtpt.numGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_DoubleMu3_DZ_PFMET90_PFMHT90_v*')
>  double_soft_muon_backup_90_mhtpt.denGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_Mu17_TrkIsoVVL_Mu8_TrkIsoVVL_DZ_v*')
>
> +#triple muon
> +triple_muon_mupt = hltTOPmonitoring.clone()
> +triple_muon_mupt.FolderName   = cms.string('HLT/SUSY/SOS/TrileMu/Muon')
> +# Selections
> +triple_muon_mupt.nmuons           = cms.uint32(3)
> +triple_muon_mupt.invMassUppercut       = cms.double(50)
> +triple_muon_mupt.invMassLowercut       = cms.double(10)
> +triple_muon_mupt.invMassCutInAllMuPairs=cms.bool(True)
> +# Binning
> +
> +# Triggers
> +triple_muon_mupt.numGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_TripleMu_5_3_3_Mass3p8to60_DZ_v*')
> +triple_muon_mupt.denGenericTriggerEventPSet.hltPaths = cms.vstring('HLT_Trimuon2_Upsilon5_Muon_v*')
>
> @mtosi <https://github.com/mtosi> I think the DoubleMuon PD seems the
> most appropriate for this path. Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/20212#discussion_r134195380>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58wukTZDZUPEiz9hZ24rJN6qh-kISks5saVmSgaJpZM4O7tLK>
> .
>

rcastello commented:
I don't see HLT_Trimuon2_Upsilon5_Muon_v anywhere in the Muon PD datasets under /cdaq/physics/Run2017/2e34/v2.2.2/HLT/V1

(only HLT_Trimuon2_Upsilon5_Muon_v but under MuOnia PD)

rcastello commented:
Sorry, I meant:

(only **_HLT_TriMuon5_3p5_2_Upsilon_Muon_** but under MuOnia PD )

mtosi commented:
can you please update the configuration w/ the proper trigger name ?

gkaratha committed:
addressing comments

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #20212 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20212/271

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20212/271/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20212/271/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

gkaratha commented:
Hi I changed everything in the new commit. Only the lepton ID I did not change. this is what we did for doublemu path... Hoiw should be changed? Thanks

mtosi commented:
are you not applying any muon ID ?
I'm surprised
 mia

On Wed, Aug 23, 2017 at 3:26 PM, gkaratha <notifications@github.com> wrote:

> Hi I changed everything in the new commit. Only the lepton ID I did not
> change. this is what we did for doublemu path... Hoiw should be changed?
> Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/20212#issuecomment-324330405>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58-KRpq30iM9jQnXRTaQU18NMsVAcks5sbCiNgaJpZM4O7tLK>
> .
>

gkaratha committed:
lepton def

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #20212 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20212/300

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20212/300/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20212/300/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

gkaratha commented:
Dear Mia,

I added globalMuon requirment. Also changed the name

mtosi commented:
I think there are some issues to be addressed (?!?)

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 98714ff6d54fedd7415f51572ff7bb88c9b54f9c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22491/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5d9892889f68e5ca315afc3b31869460844e9db2
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22491/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22491/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22491/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 246
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653499
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

gkaratha committed:
removing Pt

gkaratha commented:
Dear Mia,
We changed the mass of muon to 0.105, the name of the reference trigger, we require a globalMuons removed the mll efficiency and renamed the PR. Is it something else that I am missing? Many thanks for your help.

Best regards,
George

mtosi commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
Pull request #20212 was updated. @kmaeshima, @cmsbuild, @vanbesien, @vazzolini, @dmitrijus can you please check and sign again.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20212/342

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20212/342/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20212/342/git-diff.patch | patch -p1`

In future, you can run `scram build code-checks` to apply code checks

cmsbuild commented:
+1
Tested at: b7922a191449e6db55f3b232d51a55a9952ad9f9
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22520/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5ca2c40df047c88cd2090929074e535151529687
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22520/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22520/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20212/22520/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 205
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653540
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

davidlange6 commented on DQMOffline/Trigger/plugins/TopMonitor.cc:
```diff
@@ -460,6 +461,15 @@ void TopMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+bool TopMonitor::mllpairscut(std::vector<TLorentzVector> mu, double uppercut, double lowercut) {
+for (unsigned int idx=0; idx<mu.size(); idx++) {
+  for (unsigned int idx2=idx; idx2<mu.size(); idx2++)
+     if ((mu[idx]+mu[idx2]).M()>uppercut || (mu[idx]+mu[idx2]).M()<lowercut ) return true;
```
hi @gkaratha - surely you should not include idx=idx2 as a possibility here?

davidlange6 commented on DQMOffline/Trigger/plugins/TopMonitor.cc:
```diff
@@ -460,6 +461,15 @@ void TopMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
 
 }
+//george
+bool TopMonitor::mllpairscut(std::vector<TLorentzVector> mu, double uppercut, double lowercut) {
```
pass the vector by const &

davidlange6 commented on DQMOffline/Trigger/plugins/TopMonitor.cc:
```diff
@@ -590,6 +600,22 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
+    std::vector<TLorentzVector> mu;
+    for (unsigned int idx=0; idx<muons.size(); idx++){
+      TLorentzVector temp_mu;
+      temp_mu.SetPtEtaPhiM(muons[idx].pt(),muons[idx].eta(),muons[idx].phi(),0.105);
```
please avoid the TLorentzVector conversion. in fact, its probably best just to inline the simple calculation in mllpairscuts altogether.

gkaratha commented:
Dear @davidlange6 

I addressed the comments but in different CMSSW (the old said that is not supported anymore). Unfortunately I failed to push in this PR so I opened this: https://github.com/cms-sw/cmssw/compare/CMSSW_9_3_X...gkaratha:gkaratha/dqm_3ple_muon_sos?expand=1 This is NOT a PR yet. I wanted first to ask if there is a way to transfer everything in this PR. If not I will make the code in the link a new PR and close this .The history is preserved. Thanks

davidlange6 commented:
hi @gkaratha - maybe you should just make a new PR indeed. Thanks

gkaratha commented:
Hi @davidlange6 ,

I just did it. Should I close this? Or leave it until you check the other? Thanks
